### ISSUE TITLE
error while execute preview:pre-generate

### ISSUE THREAD

**Issue by z0ow** [nextcloud/previewgenerator#209]
My nextcloud version is NC 20.0. I found the preview generator is not working recently, so I am trying to execute `php -f /path/to/nc/occ preview:pre-generate` manually, then I get this message:

```
An unhandled exception has been thrown:
Error: Call to a member function file_get_contents() on null in [path-to-nextcloud]/lib/private/Files/Filesystem.php:716
Stack trace:
#0 [path-to-nextcloud]/lib/private/legacy/OC_Image.php(624): OC\Files\Filesystem::file_get_contents('/public/cloud/n...')
#1 [path-to-nextcloud]/lib/private/Preview/Image.php(52): OC_Image->loadFromFile('/public/cloud/n...')
#2 [path-to-nextcloud]/lib/private/Preview/GeneratorHelper.php(63): OC\Preview\Image->getThumbnail(Object(OC\Files\Node\File), 4096, 4096)
#3 [path-to-nextcloud]/lib/private/Preview/Generator.php(244): OC\Preview\GeneratorHelper->getThumbnail(Object(OC\Preview\JPEG), Object(OC\Files\Node\File), 4096, 4096)
#4 [path-to-nextcloud]/lib/private/Preview/Generator.php(140): OC\Preview\Generator->getMaxPreview(Object(OC\Files\SimpleFS\SimpleFolder), Object(OC\Files\Node\File), 'image/jpeg', '')
#5 [path-to-nextcloud]/lib/private/Preview/Generator.php(109): OC\Preview\Generator->generatePreviews(Object(OC\Files\Node\File), Array, 'image/jpeg')
#6 [path-to-nextcloud]/lib/private/PreviewManager.php(190): OC\Preview\Generator->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill', NULL)
#7 [path-to-nextcloud]/apps/previewgenerator/lib/Command/PreGenerate.php(205): OC\PreviewManager->getPreview(Object(OC\Files\Node\File), 32, 32, true)
#8 [path-to-nextcloud]/apps/previewgenerator/lib/Command/PreGenerate.php(193): OCA\PreviewGenerator\Command\PreGenerate->processFile(Object(OC\Files\Node\File))
#9 [path-to-nextcloud]/apps/previewgenerator/lib/Command/PreGenerate.php(164): OCA\PreviewGenerator\Command\PreGenerate->processRow(Array)
#10 [path-to-nextcloud]/apps/previewgenerator/lib/Command/PreGenerate.php(132): OCA\PreviewGenerator\Command\PreGenerate->startProcessing()
#11 [path-to-nextcloud]/apps/mail/vendor/symfony/console/Command/Command.php(255): OCA\PreviewGenerator\Command\PreGenerate->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 [path-to-nextcloud]/apps/mail/vendor/symfony/console/Application.php(1005): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 [path-to-nextcloud]/apps/mail/vendor/symfony/console/Application.php(271): Symfony\Component\Console\Application->doRunCommand(Object(OCA\PreviewGenerator\Command\PreGenerate), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 [path-to-nextcloud]/apps/mail/vendor/symfony/console/Application.php(147): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 [path-to-nextcloud]/lib/private/Console/Application.php(215): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 [path-to-nextcloud]/console.php(100): OC\Console\Application->run()
#17 [path-to-nextcloud]/occ(11): require_once('/home/www-data/...')
```

It looks like the preview generator is using some functions in NC but it has been removed, is it a bug in NC20?

**Comment by MichaIng** [comment]
Please try to remove and reinstall the app, at least in my case it solved this error: #204

**Comment by ZehuanZhang** [comment]
> Please try to remove and reinstall the app, at least in my case it solved this error: #204

Thanks for your solution, but it seems not working for me, even though I perform uninstall and reinstall.

**Comment by MichaIng** [comment]
Okay, our error is as well a different one. Just found it several times: #181 #145 #75 #17 
Seems like to happen with certain special characters in file name, e.g. `.AppleDouble` or with parenthesis `"` inside.

This one claims to be a fix: #191
You might try to apply this change to your `[path-to-nextcloud]/apps/previewgenerator/lib/Command/PreGenerate.php` and see if that helps.

**Comment by ZehuanZhang** [comment]
Thanks, I change the PreGenerate.php by adding the committed change as 0773308 (shown in #191). Everything is going well.

By the way, will the fix of this problem be merged into the next patch? Manually change the app's code makes me feel weird. 😢

**Comment by MichaIng** [comment]
> By the way, will the fix of this problem be merged into the next patch?

Only @rullzer does know 👋.

**Comment by joshtrichards** [comment]
Fixed in #191
